app/controllers/users_controller.rb

```diff
@@ -10,11 +10,15 @@ def update
     @user = User.find(params[:id])
     raise AuthorizationError if current_user != @user
 
-    @user.update_attributes({
-      sam_id:   params[:sam_id],
-      duns_id:  params[:duns_id]
-    })
+    @user.update_attributes(user_params)
 
     redirect_to '/'
   end
+
+
+  private
+
+  def user_params
+    params.require(:user).permit(:duns_id)
```

Comment:
I know this is a rails thing, but for me it is a big smell to have private methods in your controllers. So, I am not a fan. My style would be to keep it explicit ... also makes it easier to extract into a class later. The way you did it is the rails way though, so this is just a nudge. You can keep it as is.